### PR TITLE
AUT-584: Publish TxMA-format audit messages to SQS for consumption

### PR DESCRIPTION
**Pull request by mrwilson**
Send the MVP audit payload to TxMA's audit queue from a single endpoint in build and staging

### CODE REVIEW

File: integration-tests/src/test/java/uk/gov/di/authentication/api/UserInfoIntegrationTest.java [hunk 1]
```diff
@@ -432,5 +464,15 @@ public UserInfoConfigurationService() {
         public boolean isIdentityEnabled() {
             return true;
         }
+
+        @Override
+        public boolean isTxmaAuditEnabled() {
+            return true;
```
Comment (dbes-gds) [inline integration-tests/src/test/java/uk/gov/di/authentication/api/UserInfoIntegrationTest.java:470]: want this enabled by default given it's only going to a couple of environments?

File: shared-test/src/main/java/uk/gov/di/authentication/sharedtest/helper/AuditAssertionsHelper.java [hunk 2]
```diff
@@ -53,4 +66,32 @@ public static void assertEventTypesReceived(
...
+        await().atMost(SNS_TIMEOUT, SECONDS)
+                .untilAsserted(
+                        () ->
+                                assertThat(
+                                        tmxaAuditQueue.getApproximateMessageCount(),
+                                        equalTo(eventTypes.size())));
+
+        var receivedEvents =
+                tmxaAuditQueue.getRawMessages().stream()
+                        .peek(System.out::println)
```
Comment (dbes-gds) [inline shared-test/src/main/java/uk/gov/di/authentication/sharedtest/helper/AuditAssertionsHelper.java:86]: want to keep that?

Comment (mrwilson) [reply shared-test/src/main/java/uk/gov/di/authentication/sharedtest/helper/AuditAssertionsHelper.java:86]: Good shout, will take this out in next PR.

File: integration-tests/src/test/java/uk/gov/di/authentication/api/UserInfoIntegrationTest.java (hunk 1 above)
Comment (mrwilson) [reply integration-tests/src/test/java/uk/gov/di/authentication/api/UserInfoIntegrationTest.java:470]: True - I didn't want to duplicate every test with new configuration, but maybe that's not avoidable.

Comment (dbes-gds) [reply integration-tests/src/test/java/uk/gov/di/authentication/api/UserInfoIntegrationTest.java:470]: yes sorry that's test code

**Review by dbes-gds — APPROVED** [review]

**Pull request merged**